pkg/issuer/vault/setup.go

```diff
@@ -81,6 +83,18 @@ func (v *Vault) Setup(ctx context.Context) error {
 		return fmt.Errorf(messageAuthFieldRequired)
 	}
 
+	// check if caBundle is valid
+	certs := v.issuer.GetSpec().Vault.CABundle
+	if len(certs) > 0 {
+		caCertPool := x509.NewCertPool()
+		ok := caCertPool.AppendCertsFromPEM(certs)
+		if !ok {
+			glog.V(4).Infof("%s: %s", v.issuer.GetObjectMeta().Name, messageVaultCABundleInvalid)
+			v.issuer.UpdateStatusCondition(v1alpha1.IssuerConditionReady, v1alpha1.ConditionFalse, errorVault, messageVaultCABundleInvalid)
+			return fmt.Errorf(messageVaultCABundleInvalid)
+		}
+	}
```

Comment:
This validation should be as part of the actual API validation IMO, which will allow us to reject issuer resources with invalid CAs specified at the time they are created/updated (as opposed to when the controller is syncing the resource).

You should be able to copy this logic across to here, with some minor adjustments: https://github.com/jetstack/cert-manager/blob/master/pkg/apis/certmanager/validation/issuer.go#L115